src/operations.jl

```diff
@@ -113,7 +113,7 @@ function Base.convert(::Type{Char}, op::Operation)
 end
 
 function Base.print(io::IO, op::Operation)
-    write(io, convert(Char, op))
+    write(io, isvalid(op) ? convert(Char, op) : '?')
```

Comment:
Should we support invalid operations? Perhaps it's a better design to not allow invalid operations to exist. I'm not sure though

Reply:
The time has passed, I don't know why I've added it. Maybe when I'll be revisiting the PR, I'll see why.
Otherwise I agree, it could be removed.